### ISSUE TITLE
Character above 0xFFFF doesn't `pack` correctly

### ISSUE THREAD

**Issue by ryantrinkle** [obsidiansystems/text#9]
This program produces incorrect output, as shown in the comments below.

```
import Data.Text (Text, pack)

foreign import javascript unsafe "console['log']($1)" consoleLog :: Text -> IO ()

main :: IO ()
main = do
  let s = "\120590" -- Mathematical Italic Small Sigma https://unicode-table.com/en/1D70E/
  print s        -- Prints "\120590"
  putStrLn s     -- Shows the correct character
  let t = pack s
  print t        -- Prints "\55054" (incorrect)
  consoleLog t -- Displays instead as Hangul Syllable Hwebs https://unicode-table.com/en/D70E/
```

Thanks to @benzrf for finding this!

**Comment by ryantrinkle** [comment]
Note that the incorrect character is the same as the correct one except for the top bit being lost.

**Comment by ryantrinkle** [comment]
This was resolved by ghcjs/ghcjs-base#102.  Fix merged to reflex-platform in reflex-frp/reflex-platform#167, thanks to @luigy.
